software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+﻿script "lazydogpile"
+/*
+This stack is mostly an empty "hull" that ties everything else together
+Some stuff, like the Livecode engine and objects, need to be binary
+Everything else is in text files
+*/
+local sMain -- main[doc][rec][tag]=data/pointers
```

Comment:
Here's what I added to the file

-- sMain[1] = first DOCUMENT/DOC; theoretically there could be more, but not supported as of 16MAR2016
-- sMain[1][0] = document's RECORD/REC (primary) key for metadata
--------------[lastKey] = the highest record key used in this documentt, incremented as necessary
--------------[pointerTags] = a " " delimited list of tags in this document which contain pointer(s)
-- as of 16MAR2016 it's "step myStep turn myTurn bond prev next skim read bulk trim"
-- pointers are comma delimited & return delimited lists of numbers "#,# cr #,#"
-- the first number is the record being pointed to
-- the second number is the turn record that created this particular pointer
-- pointers are always edited in mirrored pairs, so if record 17 points to 58, 58 also points to 17
-- sMain[1][1-n] = record keys
----------------[step] = this record is pointing to all of the information tags that together make up a graph element
----------------[myStep] = points back to the step that owns a tag
----------------[turn] = this record is pointing to all of the records created by this alteration
----------------[myTurn] = points back to the turn that created this record
----------------[bond] = DEPRECATED (unnecessary)
----------------[prev] = pointer to a [next] tag backwards along a flow link
----------------[next] = pointer to a [prev] tag forwards along a flow link
----------------[skim] = pointer to a [read] tag backwards along a dive link
----------------[read] = pointer to a [skim] tag forwards along a dive link
----------------[bulk] = pointer to a [trim] tag backwards along a rise link
----------------[trim] = pointer to a [bulk] tag forwards along a rise link
----------------[act] = contains string uniquely identifying the alteration (probably the handler name, EX: newItem, newFlow, newDive, newRise)
----------------[type] = this is an information tag specifying what kind of graph element the step it's attached to is (EX: state, change, flow, dive)